### ISSUE TITLE
MSEdge / Win10 fails to download

### ISSUE THREAD

**Issue by jesstelford** [xdissent/ievms#274]
Running this command:

```bash
$ curl -s https://raw.githubusercontent.com/xdissent/ievms/master/ievms.sh | env IEVMS_VERSIONS="EDGE" bash
```

Gives me this error:

```
Downloading OVA ZIP from https://az792536.vo.msecnd.net/vms/VMBuild_20150801/VirtualBox/MSEdge/Mac/Microsoft%20Edge.Win10.For.Mac.VirtualBox.zip to MSEdge_Win10.zip (attempt 3 of 3)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   215  100   215    0     0    746      0 --:--:-- --:--:-- --:--:--   746
MD5 check failed for MSEdge_Win10.zip (wanted c1011b491d49539975fb4c3eeff16dae, got c677417b127bf7a78983072654a8390d)
Failed to download https://az792536.vo.msecnd.net/vms/VMBuild_20150801/VirtualBox/MSEdge/Mac/Microsoft%20Edge.Win10.For.Mac.VirtualBox.zip to /Users/jtelford/.ievms/MSEdge_Win10.zip (attempt 3 of 3)
```

Checking that URL in the browser gives:

```xml
<Error>
<Code>BlobNotFound</Code>
<Message>
The specified blob does not exist. RequestId:0fb87208-0001-0118-5c48-f07add000000 Time:2015-09-16T06:26:54.9299222Z
</Message>
</Error>
```

So I'm guessing the file has moved / been deleted?

**Comment by kwoodfriend** [comment]
Hit this issue as well. Modern IE sends you to https://az792536.vo.msecnd.net/vms/VMBuild_20150801/VirtualBox/MSEdge/Mac/Microsoft%20Edge.Win10.For.Mac.VirtualBox.20150801.zip

**Comment by xdissent** [comment]
Yes, it looks like it was moved. Update forthcoming.

**Comment by Jamesking56** [comment]
^ I've wrote a PR, just needs testing to see if it works :+1:

**Comment by xdissent** [comment]
Killer, will push this out today.

**Comment by Jamesking56** [comment]
Just testing it now

**Comment by Jamesking56** [comment]
@xdissent It works except it doesn't seem to have guest additions installed

**Comment by Guntrisoft** [comment]
Downloading Edge doesn't work again

Failed to download https://az792536.vo.msecnd.net/vms/VMBuild_20150801/VirtualBox/MSEdge/Mac/Microsoft%20Edge.Win10.For.Mac.VirtualBox.zip to ...

Correct URL on modern.ie seems to be this now:

https://az792536.vo.msecnd.net/vms/VMBuild_20150801/VirtualBox/MSEdge/Mac/Microsoft%20Edge.Win10.For.Mac.VirtualBox.20150801.zip

**Comment by prelator** [comment]
Was this fixed? I am still encountering this issue when I try to install Edge/Win 10.

**Comment by paulovelho** [comment]
issue still on.
you can reference the fork with the command:
curl -s https://raw.githubusercontent.com/Jamesking56/ievms/patch-1/ievms.sh | env IEVMS_VERSIONS="EDGE" bash

**Comment by Jamesking56** [comment]
If you reference my fork it works it's just missing Guest Additions in the OS

**Comment by mpassell** [comment]
@Jamesking56 I'm not familiar with what Guest Additions add to a VirtualBox VM. Does Edge still run fine in the VM (for testing purposes) without the additions?

**Comment by karlhorky** [comment]
The VM will still work without the Guest Additions, but you lose [a few things](https://www.virtualbox.org/manual/ch04.html), among those:

- shared clipboard
- shared folders
- mouse pointer integration (don't need to manually force a release of the mouse from the guest VM)

**Comment by mpassell** [comment]
Thanks @karlhorky. I'll think about whether those limitations matter to me. Even if they do, I can probably put off my Edge testing for a bit.

**Comment by jfroom** [comment]
fwiw, when using @paulovelho / @Jamesking56 forked curl, I get an error. Full log [here](https://gist.githubusercontent.com/jfroom/57a7b979f6870fa35ec1/raw/838cf01bee1f04ce79385440a33307670f713eb3/gistfile1.txt).
```
0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
Interpreting /Users/jfroom/.ievms/MSEdge - Win10.ova...
VBoxManage: error: 'Windows10_64' is not a valid Guest OS type
VBoxManage: error: Details: code NS_ERROR_INVALID_ARG (0x80070057), component VirtualBox, interface IVirtualBox, callee IAppliance
VBoxManage: error: Context: "Interpret" at line 328 of file VBoxManageAppliance.cpp
```

**Comment by Jamesking56** [comment]
@jfroom what version of Virtualbox are you using?

**Comment by jfroom** [comment]
@Jamesking56 An old one. I just upgraded to latest 4.3.30 r101610 (and installed the guest additions during upgrade). MSEdge/Win10 vm works fine now. :+1:

**Comment by Jamesking56** [comment]
Any updates on this? I'm on latest Virtualbox and still no Guest additions in the OS. However, Virtualbox's site says their version 5 guest additions supports Windows 10.

**Comment by gvlekke** [comment]
same problem here

**Comment by jsevellec** [comment]
I'm also facing this issue. Any update on this?

**Comment by cdata** [comment]
I also hit this.

**Comment by dennisgaebel** [comment]
OS: El Capitan
VB Version: 5.0.8 r103449

```bash
curl -s https://raw.githubusercontent.com/xdissent/ievms/master/ievms.sh | env IEVMS_VERSIONS="EDGE" bash
```

results in this output error stating an MD5 failure as stated by other members:

```bash
MD5 check failed for MSEdge_Win10.zip (wanted c1011b491d49539975fb4c3eeff16dae, got e4444fd16c6bd80ca56175ca027eb447)
Failed to download https://az792536.vo.msecnd.net/vms/VMBuild_20150801/VirtualBox/MSEdge/Mac/Microsoft%20Edge.Win10.For.Mac.VirtualBox.zip to /Users/GrayGhost/.ievms/MSEdge_Win10.zip (attempt 3 of 3)
```

NOTE: My ``.ievms`` folder is completely empty before making the install.

**Comment by simbo1905** [comment]
Had the problem when with the PR mentioned above which was 

```
curl -s https://raw.githubusercontent.com/Jamesking56/ievms/patch-1/ievms.sh | env IEVMS_VERSIONS="EDGE" bash
```

I donated $5 towards getting this fixed.

**Comment by maxfenton** [comment]
Yeah, still hitting a 404 at `https://az792536.vo.msecnd.net/vms/VMBuild_20150801/VirtualBox/MSEdge/Mac/Microsoft%20Edge.Win10.For.Mac.VirtualBox.zip` — thanks for the patches, people.

**Comment by streamtree** [comment]
The problem seems to be that MS is no longer making Mac-specific versions of the files. (To verify, look at the links used to download the "Mac" versions; they're the same as the links for the Windows versions). To fix, replace
```
https://az792536.vo.msecnd.net/vms/VMBuild_20150801/VirtualBox/MSEdge/Mac/Microsoft%20Edge.Win10.For.Mac.VirtualBox.zip
```
with
```
https://az792536.vo.msecnd.net/vms/VMBuild_20150801/VirtualBox/MSEdge/Windows/Microsoft%20Edge.Win10.For.Windows.VirtualBox.zip
```

**Comment by maxfenton** [comment]
:+1:

**Comment by karlhorky** [comment]
@streamtree Can these changes be made into a pull request?

Changing the URL and the MD5 sound like easy tasks, but the manual import - can this also be fixed in code?

**Comment by abstracthat** [comment]
@karlhorky Following up @streamtree 's comment you can rename the file that got extracted from the ZIP... rename from `IE11 - Win10.ova` to `MSEdge - Win10.ova`. Then run the script again and it should complete the import/set up for you. Working great here. I'm not sure what all would be involved in a PR for this, but hopefully this helps.

**Comment by mysterycommand** [comment]
Can @cdata's fork/work be merged in? Would be great to have this "just work" … well, almost (Microsoft's new slogan?).

**Comment by cdata** [comment]
@mysterycommand I will make a PR, although I haven't checked to see if this script is still valid in the last few days ;)

**Comment by niksy** [comment]
I suppose file rename is also necessary?

**Comment by cdata** [comment]
@niksy Yes, I was going to add that in, but did not get around to it. I will try to do it later today.

**Comment by xthilakx** [comment]
Ran into the same issue today 

````
MD5 check failed for MSEdge_Win10.zip (wanted c1011b491d49539975fb4c3eeff16dae, got a4698907893b267b2c690741e40edb30)
Check failed - redownloading OVA ZIP
Downloading OVA ZIP from https://az792536.vo.msecnd.net/vms/VMBuild_20150801/VirtualBox/MSEdge/Mac/Microsoft%20Edge.Win10.For.Mac.VirtualBox.zip to MSEdge_Win10.zip (attempt 3 of 3)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   215  100   215    0     0    273      0 --:--:-- --:--:-- --:--:--   273
MD5 check failed for MSEdge_Win10.zip (wanted c1011b491d49539975fb4c3eeff16dae, got 7e14bb21bafb7f91f19fa473813d0aaa)
Failed to download https://az792536.vo.msecnd.net/vms/VMBuild_20150801/VirtualBox/MSEdge/Mac/Microsoft%20Edge.Win10.For.Mac.VirtualBox.zip to /Users/thilak/.ievms/MSEdge_Win10.zip (attempt 3 of 3)
`````

Looks like the file has been moved again. Can we reopen this issue?

**Comment by barillax** [comment]
Running into this issue today again.

**Comment by hiddentao** [comment]
Me too, can't get EDGE.

**Comment by xdissent** [comment]
Thanks @streamtree, downloading now to make sure there's no surprises then I'll update.

**Comment by airblade** [comment]
I got Edge working with two steps from above:

- @steamtree's [comment](https://github.com/xdissent/ievms/issues/274#issuecomment-159412949)
- @abstracthat's [comment](https://github.com/xdissent/ievms/issues/274#issuecomment-161167031)

**Comment by xthilakx** [comment]
I don't care enough about Edge right now, so I can live without it, but if you folks got it work, please send a pull request.

**Comment by xdissent** [comment]
Thanks @airblade. I got it running this weekend but the OVA name is a quirk I wasn't expecting. I've got to decide whether to rename it or change the import, then I'll push a release. Everything else about the windows version seems to be ok.

**Comment by vedranjaic** [comment]
Same error here, cannot get EDGE

**Comment by maticrivo** [comment]
@xdissent any news on this?

**Comment by benross** [comment]
Hey @xdissent  and @airblade  any update with this? Us n00bs that are looking for a one-line download of edge are still waiting for this as far as I know.

**Comment by Jamesking56** [comment]
Hey everyone, Sorry I've been so busy (I've just moved house). I'm going to create another PR for this to see if I can fix it.

**Comment by Jamesking56** [comment]
Interestingly enough, if you download the Edge VM from modern.ie and extract it manually, it gives you a virtual machine called 'IE11 - Win10'... which is wrong.

**Comment by Jamesking56** [comment]
See my PR for installing Edge ^

**Comment by benross** [comment]
Thanks @Jamesking56...works like a charm!

**Comment by tannerhodges** [comment]
Awesome, thanks @Jamesking56.

**Comment by kenbird** [comment]
@Jamesking56 I just tried running the commands from ur PR, it looks like even though I'm requesting EDGE I'm receiving IE11....and have received this error:

`$ curl -s https://raw.githubusercontent.com/Jamesking56/ievms/patch-2/ievms.sh | env IEVMS_VERSIONS="EDGE" bash
Checking for VirtualBox
Checking for Oracle VM VirtualBox Extension Pack
Pack no. 0:   Oracle VM VirtualBox Extension Pack
Building IE EDGE VM
Checking for existing OVA at /Users/me/.ievms/IE11 - Win10.ova
Downloading OVA ZIP from https://az792536.vo.msecnd.net/vms/VMBuild_20150801/VirtualBox/MSEdge/Windows/Microsoft%20Edge.Win10.For.Windows.VirtualBox.zip to IE11_Win10.zip (attempt 1 of 3)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 5084M  100 5084M    0     0  6828k      0  0:12:42  0:12:42 --:--:-- 7052k
MD5 check succeeded for IE11_Win10.zip
Extracting OVA from /Users/me/.ievms/IE11_Win10.zip
IE11_Win10.zip: Zip
  IE11 - Win10.ova  (5384994816 B)... OK.
Successfully extracted to "./IE11 - Win10.ova".
Checking for existing MSEdge - Win10 VM
Creating MSEdge - Win10 VM (disk: /Users/me/.ievms/MSEdge - Win10-disk1.vmdk)
0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
Interpreting /Users/me/.ievms/IE11 - Win10.ova...
VBoxManage: error: 'Windows10_64' is not a valid Guest OS type
VBoxManage: error: Details: code NS_ERROR_INVALID_ARG (0x80070057), component VirtualBox, interface IVirtualBox, callee IAppliance
VBoxManage: error: Context: "Interpret" at line 328 of file VBoxManageAppliance.cpp`

**Comment by tjgfernandes** [comment]
+1 thanks @Jamesking56

**Comment by PixelHeroMedia** [comment]
This is failing still for Edge Windows 10 download!

```
Check failed - redownloading OVA ZIP
Downloading OVA ZIP from https://az792536.vo.msecnd.net/vms/VMBuild_20150801/VirtualBox/MSEdge/Mac/Microsoft%20Edge.Win10.For.Mac.VirtualBox.zip to MSEdge_Win10.zip (attempt 3 of 3)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   215  100   215    0     0    777      0 --:--:-- --:--:-- --:--:--   778
MD5 check failed for MSEdge_Win10.zip (wanted c1011b491d49539975fb4c3eeff16dae, got a5a5da8aa2fb4523a21e1897e070c79c)
Failed to download https://az792536.vo.msecnd.net/vms/VMBuild_20150801/VirtualBox/MSEdge/Mac/Microsoft%20Edge.Win10.For.Mac.VirtualBox.zip to /Users/richard.edwards/.ievms/MSEdge_Win10.zip (attempt 3 of 3)
```
